### DISCUSSION TITLE
Turn a Query object back into a standard SQL

### DISCUSSION THREAD

**Discussion by calebeaires** [paustint/soql-parser-js#170] [Ideas]
this lib is amazing. make relationships with object graph-like notation is great. Do you consider including a function to turn a query object back into a **Standard SQL** so we can use it with MySQL or even Postgres?

**Comment by paustint** [comment]
@calebeaires - I had not considered that use-case, but I probably would have a use for turning it into a "big query" sql query.

I am not an SQL expert, so it would take me some time to learn and work through the details - but I will try to keep this ida in mind when I get bored and want to work on something new :D

If you have any details about a specific use-case or what type of options might need to be available, I would love to hear about them.

The integration I saw where SFDC was copied into BigQuery, all the object and field names were modified - I think double underscores were turned into one underscore or something like that, so I think that an option for some basic transformation or mapping might be required to support use-cases like this.

**Reply by calebeaires** [reply]
Almost al is the same. Basic the relationships becomes left outer join. I will take a look at that. If you do some progress let me know!
